haskell/private/actions/link.bzl

```diff
@@ -520,36 +559,14 @@ def link_library_dynamic(hs, cc, dep_info, extra_srcs, objects_dir, my_pkg_id):
     else:
         dynamic_library_tmp = dynamic_library
 
-    link_ctx = dep_info.cc_dependencies.dynamic_linking
-    trans_link_ctx = dep_info.transitive_cc_dependencies.dynamic_linking
-
-    # Direct C library dependencies to link.
-    # I.e. not indirect through another Haskell dependency.
-    # Such indirect dependencies are linked by GHC based on the extra-libraries
-    # fields in the dependency's package configuration file.
-    libs_to_link = get_mangled_libs(link_ctx.libraries_to_link.to_list())
-    import_libs_to_link = set.to_list(dep_info.import_dependencies)
-    _add_external_libraries(args, libs_to_link + import_libs_to_link)
-
-    # Transitive library dependencies to have in scope for linking.
-    trans_libs_to_link = get_mangled_libs(trans_link_ctx.libraries_to_link.to_list())
-    trans_import_libs = set.to_list(dep_info.transitive_import_dependencies)
-
-    if hs.toolchain.is_darwin:
-        _fix_darwin_linker_paths(
-            hs,
-            dynamic_library_tmp,
-            dynamic_library,
-            trans_link_ctx.libraries_to_link.to_list() + trans_import_libs,
-        )
-
-    # Direct dynamic library dependencies to have in RUNPATH.
-    cc_solibs = trans_link_ctx.dynamic_libraries_for_runtime.to_list()
-
-    # Configure RUNPATH.
-    solibs = cc_solibs + trans_import_libs + set.to_list(dep_info.dynamic_libraries)
-    for rpath in set.to_list(_infer_rpaths(hs.toolchain.is_darwin, dynamic_library, solibs)):
-        args.add("-optl-Wl,-rpath," + rpath)
+    (cc_link_libs, _cc_solibs) = _link_dependencies(
+        hs,
+        dep_info,
+        True,  # dynamic linking mode
```

Comment:
You can use named arguments in invocations in skylark, `dynamic = True`.

I’d strongly suggest we switch all calls to explicitly name arguments.